src/font/SharedGrid.zig

```diff
@@ -332,11 +332,12 @@ const GlyphKey = struct {
 
     const Context = struct {
         pub fn hash(_: Context, key: GlyphKey) u64 {
-            return @bitCast(Packed.from(key));
+            const packed_key = Packed.from(key);
+            return std.hash.CityHash64.hash(std.mem.asBytes(&packed_key));
         }
 
         pub fn eql(_: Context, a: GlyphKey, b: GlyphKey) bool {
-            return Packed.from(a) == Packed.from(b);
+            return a.glyph == b.glyph and Packed.from(a) == Packed.from(b);
```

Comment:
This makes sense. It's slower in the equal case but faster in the unequal case which is more common, presumably?

Reply:
yes, the overwhelmingly common case is the glyph's are different and we avoid packing both. It could be further unrolled like I did with the Style eql, but that didn't have such an massive and fast u32 cmp common case.